### PR TITLE
fix: update message content border radius

### PR DESCRIPTION
**Pull request by Enigma-I-am**
## 🎯 Goal
This PR gives the customer the ability to set the border-radius of the message content components. This PR also resolves this [issue](https://github.com/GetStream/stream-chat-react-native/issues/1182)

## 🛠 Implementation details
The required implementation for the styles to be implemented on the top-left, top-right borders and also borderRadius to update all borders have been added in MessageContent.tsx.

## 🎨 UI Changes

<details>
<summary>iOS</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

<details>
<summary>Android</summary>

<table>
    <thead>
        <tr>
            <td>Before</td>
            <td>After</td>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>
                
            </td>
            <td>
                
            </td>
        </tr>
    </tbody>
</table>
</details>

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) spec
- [ ] New code is covered by tests
- [x] Screenshots added for visual changes
- [ ] Documentation is updated

### CODE REVIEW

File: package/src/components/Message/MessageSimple/MessageContent.tsx [hunk 1]
```diff
@@ -222,6 +222,19 @@ const MessageContentWithContext = <
   }
 
   const repliesCurveColor = isMyMessage && !error ? backgroundColor : grey_whisper;
+  const isBorderBottomLeftRadius =
```
Comment (vanGalilea) [inline package/src/components/Message/MessageSimple/MessageContent.tsx:225]: DRY?

File: package/src/components/Message/MessageSimple/MessageContent.tsx [hunk 2]
```diff
@@ -297,17 +310,11 @@ const MessageContentWithContext = <
...
-                (groupStyle === 'right_bottom' || groupStyle === 'right_single') &&
-                (!hasThreadReplies || threadList)
-                  ? borderRadiusS
-                  : borderRadiusL,
-              borderColor: isMyMessage && !error ? backgroundColor : grey_whisper,
+              borderBottomLeftRadius: isBorderBottomLeftRadius ? borderRadiusS : borderRadiusL,
+              borderBottomRightRadius: isBorderBottomRightRadius ? borderRadiusS : borderRadiusL,
+              borderColor: isBorderColor ? backgroundColor : grey_whisper,
+              borderTopLeftRadius: isBorderTopLeftRadius ? borderRadiusS : borderRadiusL,
+              borderTopRightRadius: isBorderTopRightRadius ? borderRadiusS : borderRadiusL,
```
Comment (vanGalilea) [inline package/src/components/Message/MessageSimple/MessageContent.tsx:317]: Discussion: let's also give the option to change everything at the same time via `borderRadius` ?

**Comment by madsroskar** [comment]
Can you change the PR title to reflect better what the changes in the PR actually _do_ rather than referring to the issue it seeks to fix? We won't have that context in X amount of time while looking through the commit history of the repo

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,4 @@ NEXT_RELEASE_CHANGELOG.md
 artifacts
 e2e/.env
 *.log
-.DS_STORE
\ No newline at end of file
+.DS_STORE
```
Comment (madsroskar) [inline .gitignore:6]: This change is in two different PRs simultaneously, which could cause conflicts. Not a biggie most likely in this case, but it's usually easier to separate that out to its own PR if the same happens in the future.

File: package/src/components/Message/MessageSimple/MessageContent.tsx (hunk 1 above)
Comment (madsroskar) [reply package/src/components/Message/MessageSimple/MessageContent.tsx:225]: To expand a bit on this - it seems like the same thing is happening multiple times/the same code is copied with the only difference being the group style strings. Could you refactor it so we're not repeating ourselves as much?

**Review by madsroskar — CHANGES_REQUESTED** [review]

File: package/src/components/Message/MessageSimple/MessageContent.tsx (hunk 2 above)
Comment (Enigma-I-am) [reply package/src/components/Message/MessageSimple/MessageContent.tsx]: This makes sense

File: package/src/components/Message/MessageSimple/MessageContent.tsx [hunk 4]
```diff
@@ -297,17 +303,21 @@ const MessageContentWithContext = <
...
-                  ? borderRadiusS
-                  : borderRadiusL,
-              borderBottomRightRadius:
-                (groupStyle === 'right_bottom' || groupStyle === 'right_single') &&
-                (!hasThreadReplies || threadList)
-                  ? borderRadiusS
-                  : borderRadiusL,
-              borderColor: isMyMessage && !error ? backgroundColor : grey_whisper,
+
+              borderBottomLeftRadius: shouldApplyBorderRadius('left_bottom', 'left_single')
```
Comment (vanGalilea) [inline package/src/components/Message/MessageSimple/MessageContent.tsx:307]: there is still unneeded duplication, let's just calculate and return either borderRadiusS or borderRadiusL

**Review by vanGalilea — CHANGES_REQUESTED** [review]

**Review by vanGalilea — APPROVED** [review]

**Review by madsroskar — APPROVED** [review]

**Comment by khushal87** [comment]
Hey, @vanGalilea @madsroskar was this PR tested on an android/iOS device on your end before merge?

**Pull request merged**